docs/guides/server/fips.adoc

```diff
@@ -131,18 +131,22 @@ KC(BCFIPS version 1.000203 Approved Mode, FIPS-JVM: enabled) version 1.0 - class
 supported in {project_name} when using strict mode. Some examples are importing or generating a keystore of an OIDC or SAML client in the Admin Console or for a `java-keystore` provider in the realm keys.
 
 * User passwords must be 14 characters or longer. {project_name} uses PBKDF2 based password encoding by default. BCFIPS approved mode requires passwords to be at least 112 bits
-(effectively 14 characters) with PBKDF2 algorithm. If you want to allow a shorter password, set the property `max-padding-length` of provider `pbkdf2-sha256` of SPI `password-hashing`
+(effectively 14 characters) with PBKDF2 algorithm. If you want to allow a shorter password, set the property `max-padding-length` of provider `pbkdf2-sha512` of SPI `password-hashing`
 to value 14 to provide additional padding when verifying a hash created by this algorithm. This setting is also backwards compatible with previously stored passwords.
 For example, if the user's database is in a non-FIPS environment and you have shorter passwords and you want to verify them now with {project_name} using BCFIPS in approved mode, the passwords should work.
 So effectively, you can use an option such as the following when starting the server:
 
 [source]
 ----
---spi-password-hashing-pbkdf2-sha256-max-padding-length=14
+--spi-password-hashing-pbkdf2-sha512-max-padding-length=14
 ----
 
 NOTE: Using the option above does not break FIPS compliance. However, note that longer passwords are good practice anyway. For example, passwords auto-generated by modern browsers match this
-requirement as they are longer than 14 characters.
+requirement as they are longer than 14 characters. If you want to omit the option for max-padding-length, you can set password policy to your realms to have passwords at least 14 characters long.
+
+NOTE: When you are migrating from {project_name} older than 24, or if you explicitly set the password policy to override the default hashing algorithm, it is possible that some of your users use an older
+algorithm like `pbkdf2-sha256`. In this case, it may be good to add also the option `--spi-password-hashing-pbkdf2-sha256-max-padding-length=14` to make sure that users having their passwords hashed with
+the older `pbkdf2-sha256` are able to login as well in case that their passwords are shorter than 14 characters.
```

Comment:
```suggestion
algorithm like `pbkdf2-sha256`. In this case, consider adding  the `--spi-password-hashing-pbkdf2-sha256-max-padding-length=14` option to ensure that users having their passwords hashed with
the older `pbkdf2-sha256` can log in because their passwords may be shorter than 14 characters.
```